File: js/hello.js
/**
 * @author SkilStak Coding Arts <learn@skilstak.com>
 * @copyright 2015 SkilStak, Inc.
 * @license {link http://unlicense.org/|UNLICENSE}
 *
 * @overview
 *
 * Not yo grandma's hello world.
 *
 * This program can be run from the command line (see
 * <code>parseArgs()</code>) or imported as a module
 * <code>var hello = require("hello") so
 * that other programs can use its functions.
 * 
 * This documentation can be displayed with ``pydoc3 ./hello``
 * or converted to web page with any number of tools.
 * 
 */

'use strict';

var out = process.stdout; // instead of console.log()

module.exports = {

  printPlain: function (message) {
    out.write(c.clear + message + "\n");  
  },

  printColor: function (message) {
    while (true) {
      out.write(c.rc() + message + ' ' + c.x);
    }
  },

  printMulti: function (message) {
    setInterval(function () { 
      out.write(c.clear + c.multi(message + " "));
      out.write("\n");
    }, 500);
  },

  parseArgs: function () {
    var who = "world";
    var option = "";
    if (process.argv.length > 3) {
      option = process.argv[2];
      who = process.argv[3];
    } else if (process.argv.length == 3) {
      if (process.argv[2][0] == "-") {
        option = process.argv[2];
      } else {
      who = process.argv[2];
      }
    }
    return {"who":who, "option": option};
  }
}

File: py/hello.py
"""Not yo grandma's hello world module.

This module is a collection of functions that can be called
from programs or other modules that import it. 

Notice there is no shebang line because there is no ``if __name__
== '__main__'`` section in this module, although it could have one
for testing, etc.

This documentation can be displayed with ``pydoc3 hello``
or converted to web page with any number of tools.

"""

import skilstak.colors as c
import time

def print_plain(message):
    """Prints <message> to the screen.

    Args:
        message (str): message to print 

    """
    print(message)

def print_stripes(message):
    """ Prints <message> to the screen forever as stripes

    Args:
        message (str): message to print

    """
    while True:
        print(message,end="                         ")

def print_multi(message):
    """Prints <message> in multiple colors like a Christmas tree (or Vegas sign).

    Args:
        message (str): message to print 

    """
    while True:
        print(c.clear + c.multi(message))
        time.sleep(0.5)

def print_color(message):
    """Fills the screen with randomly colored <message>s.
    
    Args:
        message (str): message to print 

    """
    print(c.clear)
    while True:
        print(c.rc() + message + c.reset, end=" ")

# This one is just for fun. Invent your own!
def print_merica(message):
    """Fills the screen with politically correctly colored <message>.
    
    Args:
        message (str): message to print 

    """
    print(c.clear)
    while True:
        print(c.merica(message), end=" ")

def parse_args():
    """Parse and return properties from the command line.

    Returns:
        dict: A dictionary of properties (name, option)

    """
    # import here instead of top since only this function uses
    from sys import argv

    name = "world"
    option = ""

    if len(argv) == 2:
        if argv[1].startswith("-"):
            option = argv[1]
        else:
            name = argv[1]
    elif len(argv) > 2:
        option = argv[1]
        name = argv[2]

    return name, option

